backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/impl/AbstractElasticsearchIndexFieldTypeConverterContext.java

```diff
@@ -56,11 +56,18 @@ final ElasticsearchIndexFieldTypeBuildContext getBuildContext() {
 	}
 
 	final FromDocumentFieldValueConverter<? super F, ?> createIndexToProjectionConverter() {
-		/*
-		 * TODO HSEARCH-3512 Throw an exception when the field is non projectable due to the bridge.
-		 * E.g. if the field is bridged and the bridge did not specify a "FromIndexFieldValueConverter".
-		 */
-		return indexToProjectionConverter == null ? createFromDocumentRawConverter() : indexToProjectionConverter;
+		if ( indexToProjectionConverter != null ) {
+			// this case corresponds to a two-way bridge
+			return indexToProjectionConverter;
+		}
+
+		if ( dslToIndexConverter != null ) {
```

Comment:
Again... the DSL converter is completely unrelated and should not be mentioned here. This entire method can be replaced with:

```
return indexToProjectionConverter;
```

That's all.